engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/URLValidatorTest.java

```diff
@@ -290,10 +290,91 @@ private void runUrlContainerValidation(Validator validator, URLContainer contain
+		assertTrue( validator.isValid( "http://userid:password@example.com/", null ) );
+		assertTrue( validator.isValid( "http://142.42.1.1/", null ) );
+		assertTrue( validator.isValid( "http://142.42.1.1:8080/", null ) );
+		assertTrue( validator.isValid( "http://➡.ws/䨹", null ) );
+		assertTrue( validator.isValid( "http://⌘.ws", null ) );
+		assertTrue( validator.isValid( "http://⌘.ws/", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_(wikipedia)#cite-1", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_(wikipedia)_blah#cite-1", null ) );
+		assertTrue( validator.isValid( "http://foo.com/unicode_(✪)_in_parens", null ) );
+		assertTrue( validator.isValid( "http://foo.com/(something)?after=parens", null ) );
+		assertTrue( validator.isValid( "http://☺.damowmow.com/", null ) );
+		assertTrue( validator.isValid( "http://code.google.com/events/#&product=browser", null ) );
+		assertTrue( validator.isValid( "http://j.mp", null ) );
+		assertTrue( validator.isValid( "ftp://foo.bar/baz", null ) );
+		assertTrue( validator.isValid( "http://foo.bar/?q=Test%20URL-encoded%20stuff", null ) );
+		assertTrue( validator.isValid( "http://مثال.إختبار", null ) );
+		assertTrue( validator.isValid( "http://例子.测试", null ) );
+		assertTrue( validator.isValid( "http://उदाहरण.परीक्षा", null ) );
+		assertTrue( validator.isValid( "http://-.~_!$&'()*+,;=:%40:80%2f::::::@example.com", null ) );
+		assertTrue( validator.isValid( "http://1337.net", null ) );
+		assertTrue( validator.isValid( "http://a.b-c.de", null ) );
+		assertTrue( validator.isValid( "http://223.255.255.254", null ) );
+
+		// invalid urls:
 		assertFalse( validator.isValid( "http", null ) );
 		assertFalse( validator.isValid( "ftp//abc.de", null ) );
-		assertTrue( validator.isValid( "ftp://abc.de", null ) );
+		assertFalse( validator.isValid( "//", null ) );
+		assertFalse( validator.isValid( "//a", null ) );
+		assertFalse( validator.isValid( "///", null ) );
+		assertFalse( validator.isValid( "///a", null ) );
+		assertFalse( validator.isValid( "foo.com", null ) );
+		assertFalse( validator.isValid( ":// should fail", null ) );
+//		assertFalse( validator.isValid( "rdar://1234", null ) );
+//		assertFalse( validator.isValid( "ftps://foo.bar/", null ) );
+//		assertFalse( validator.isValid( "h://test", null ) );
+
+		if ( validator instanceof RegexpURLValidator ) {
+//			assertFalse( validator.isValid( "http://", null ) );
+//			assertFalse( validator.isValid( "http://.", null ) );
+//			assertFalse( validator.isValid( "http://..", null ) );
+//			assertFalse( validator.isValid( "http://../", null ) );
+//			assertFalse( validator.isValid( "http://?", null ) );
+//			assertFalse( validator.isValid( "http://??", null ) );
+//			assertFalse( validator.isValid( "http://??/", null ) );
+//			assertFalse( validator.isValid( "http://#", null ) );
+//			assertFalse( validator.isValid( "http://##", null ) );
+//			assertFalse( validator.isValid( "http://##/", null ) );
```

Comment:
All the ones above are indeed an issue.

If we decide to fix this, we might do it by using the domain validation we apply in EmailValidator to the domain part. I think it should fix most of the issues.